microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,153 @@
+
+The combination of SmallRye JWT with Elytron JWT will be the pair of projects responsible for providing the functionality described in the MicorProfile JWT specification combined with tying in the integration to the application server's security framework to ensure aspects such as identity association and establishment are correctly performed.
+
+Once an identity has been established according to the steps described by the MicroProfile JWT specification the remaining security decisions required are covered by the relevant Jakarta EE specifications in use by the various subsystems. 
+
+=== Nice-to-Have Requirements
+
+This section is deliberately empty, the intent of this enhancement is to integrate MicroProfile JWT support to WildFly.  Once the initial integration is in place nice to have functionality can be added under individual enhancements with an appropriately constrained scope.
+
+=== Non-Requirements
+
+This integration will be making use of WildFly Elytron exclusively, no support will be added for the deprecated PicketBox project.  Having said that the underlying security framework will be transparent for deployments developed according to the specification - the framework only becomes apparent if there is a desire to use vendor specific APIs.
+
+This integration will not be supporting the use of pre-configured SecurityDomains, the general motivation to use MicroProfile JWT is to use an authentication mechanism where identities are self described using the tokens exchanged eliminating the need for configured security resources for the secured endpoint(s).  Use cases could be identified where pre-configured resources add an additional benefit but those would be reviewed under their own enhancements.
+
+It is assumed when working with the MicroProfile specifications that we will be supporting a single deployment, this deployment may contain sub-deployments.  The SecurityDomain of each top level deployment will be isolated from the SecurityDomain of any parallel top level deployment.  Advanced WildFly Elytron features such identity propagation across domains / top level deployments will not be supported.  At a later point we could consider an enhancement to offer more control of the SecurityDomains in use but that is out of scope for this enhancement.
+
+For this initial implementation propagation of identities is outside the current scope.  The specification does make provision to access the token passed to the server so deployments can manually retrieve a token.  Under WFLY-11868 enhancements are being provided to make use of the WildFly Elytron authentication client in RESTEasy, once available under a new RFE the microprofile-jwt subsystem can dynamically configure an `AuthenticationContext` configuration to automate propagation.  This could also introduce a new feature to the client to allow a feature to convert it's current configuration to obtain an alternative token.
```

Comment:
How would the identity propagation look in JWT context?

Reply:
Identify propagation would be along the lines of re-using the token that we received for an ongoing call to another endpoint on another server.

For now the MicroProfile JWT specification already provides access to the token so any deployments can do this manually, the point we may consider in the future is the automation of this.